### PR TITLE
fix: explicitly pass token

### PR DESCRIPTION
**Pull request by ColinBuyck**
# Pull Request Template

## Issue Overview

This PR addresses #1640 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This is a more long-term fix to the password reset and account confirmation issues. By passing the token explicitly, the userControllerProfile() stops throwing an error. This PR also fixes an issue where the isUserConfirmationTokenValid was being called multiple times and falsely triggers the link expired modal.

## How Can This Be Tested/Reviewed?

This can be tested by going through the password reset and new user confirmation on both the public and partner's site.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Review by emilyjablonski — APPROVED** [review]
🧀

File: shared-helpers/src/AuthContext.ts [hunk 1]
```diff
@@ -317,12 +319,13 @@ export const AuthProvider: FunctionComponent<React.PropsWithChildren> = ({ child
...
           dispatch(saveToken({ accessToken: response.accessToken, apiUrl, dispatch }))
-          // 12/18 - Short-term error fix to allow user accesss
-          // const profile = await userService?.userControllerProfile()
-          // if (profile) {
-          //   dispatch(saveProfile(profile))
-          //   return profile
-          // }
+          const profile = await userService?.userControllerProfile({
+            headers: { Authorization: `Bearer ${response.accessToken}` },
+          })
```
Comment (seanmalbert) [inline shared-helpers/src/AuthContext.ts:324]: Suggestion (non-blocking): It looks like we have this same bit of code four times in this file. A function could be created that accepts a 'token' parameter and returns the profile.

File: sites/partners/src/pages/reset-password.tsx
Comment (seanmalbert) [inline sites/partners/src/pages/reset-password.tsx:1]: Suggestion (non-blocking): create an issue for detroit and core that creates a reset-password component in shared-helpers with props and logic to handle the minor differences between this component in public and partners.

**Review by seanmalbert — APPROVED** [review]
I have a couple of non-blocking suggestions, otherwise looks good, @ColinBuyck !

File: shared-helpers/src/AuthContext.ts (hunk 1 above)
Comment (ColinBuyck) [reply shared-helpers/src/AuthContext.ts:324]: Great point! I'll do this in a follow-up to reduce the changes here and give us the most time to test.

**Pull request merged**